k9mail/src/main/java/com/fsck/k9/mailstore/LockableDatabase.java

```diff
@@ -380,6 +372,7 @@ private void openOrCreateDataspace() throws UnavailableStorageException {
             } catch (SQLiteException e) {
                 // try to gracefully handle DB corruption - see issue 2537
                 Log.w(K9.LOG_TAG, "Unable to open DB " + databaseFile + " - removing file and retrying", e);
+                // noinspection ResultOfMethodCallIgnored, not much we can do if this fails
```

Comment:
What are the chances of `doOpenOrCreateDb()` succeeding if you weren't able to delete the file and `doOpenOrCreateDb()` has failed before? We should fail early. So yes, I think we should throw an exception when the file couldn't be deleted. The warning is reminder enough for me, but a TODO comment certainly can't hurt. That still doesn't mean it's okay to ignore the warning :)